In [207]:
import pandas as pd
import numpy as np

import scipy.stats as stats
import statsmodels.api as sm
from statsmodels.formula.api import ols
import matplotlib.pyplot as plt


In [208]:
# loading our dataframes
#we will be using df_m for our sample of math class studetns and df_p for our sameple of portugees class. 

df_m = pd.read_csv('../data/student-mat.csv', delimiter=';')
df_p = pd.read_csv('../data/student-por.csv', delimiter=';')

Before we start our statistical analysis, we need to determine 2 things, hypothesis (H1) and statisitcal test we can use to reject or fail to reject our null hypothesis (H0)

Hypothesis: 
    The main hypothesis is to see is there any effect of being in a relationship on the final grade for the school.  

Therefore: 

    H0: There is no main effect of relationship status on the final grades of the students.
    H1: There is a main effect of relationship status on the final grades of the students. 

Note: 
    From our exploration stage, we identify 1 dependent variable (quantitative variable that we want to compare across different groups) 1 independet variables (categorical variables that define our groups)
        Dependet variable - final grade (G3 column) 
        Independet varaible - Relationship status (singe or taken)

Statistical Test: 
    The perfect statistical test in our case is One-way ANOVA

Our new Hypothesis are: 
    H0: There is no main effect of relationship status on the final grades of the students.
    H1: There is a main effect of relationship status on the final grades of the students.

In [209]:
#For our test we need to create a model where we will specify our variables 
model_m = ols ('G3 ~ C(romantic)', data = df_m).fit()
model_p = ols ('G3 ~ C(romantic)', data = df_p).fit()

In [210]:
#Time to calculate and see our resutls for our two-way ANOVA TEST 
anova_m = sm.stats.anova_lm(model_m, typ=1)
anova_p = sm.stats.anova_lm(model_p, typ=1)

print('One-way ANOVA TEST results for Math class:')
print(round(anova_m, 4))
print(' ')
print('One-way ANOVA TEST results for Portugeese class')
print(round(anova_p, 4))

One-way ANOVA TEST results for Math class:
                df     sum_sq   mean_sq       F  PR(>F)
C(romantic)    1.0   139.6969  139.6969  6.7527  0.0097
Residual     393.0  8130.2120   20.6876     NaN     NaN
 
One-way ANOVA TEST results for Portugeese class
                df     sum_sq  mean_sq       F  PR(>F)
C(romantic)    1.0    55.4944  55.4944  5.3527   0.021
Residual     647.0  6707.7722  10.3675     NaN     NaN


Let's make sense of the results for our two-way ANOVA test

For Math Class
    Our P values are:
        Romance and Grades: we REJECT the H0 because p-value is 0.0097 and our alpha score is 0.05. So p-value > a. 
            Therfoe we can conlcude: 
                There is a main effect of relationship status on the final grades of the students. 

    Conclusion: There is a main effect of relationship status on the final grades of the students in Math Class. 


For Portugeese class
    Our P values are:
        Romance and Grades: we REJECT the H0 because p-value is 0.021 and our alpha score is 0.05. So p-value < a. 
            Therfoe we can conlcude: 
                There is a main effect of relationship status on the final grades of the students. 

    Conclusion: There is a main effect of relationship status on the final grades of the students in Portugees class. 